### PR TITLE
Feature/moar tests

### PR DESCRIPTION
**Pull request by st3v3nhunt**

### CODE REVIEW

File: test/integration/bankHolidayAlert.js [hunk 1]
```diff
@@ -49,20 +49,19 @@ describe('The bank holiday alert messaging', () => {
         .times(1)
         .reply(200, serviceApiResponse);
 
-      chai.request(server)
+      return chai.request(server)
```
Comment (neilbmclaughlin) [inline test/integration/bankHolidayAlert.js]: The use of promises in the tests looks good. Some of the chai.request() statements are returned and others not and I can't figure out why. Could you elaborate please.

Comment (st3v3nhunt) [reply test/integration/bankHolidayAlert.js]: They are all returned, just some are done implicitly.
https://eslint.org/docs/rules/arrow-body-style

Comment (st3v3nhunt) [reply test/integration/bankHolidayAlert.js]: Returning the promise is required https://mochajs.org/#working-with-promises

File: test/integration/app.js [hunk 2]
```diff
@@ -10,25 +10,24 @@ const expect = chai.expect;
...
-        iExpect.htmlWith200Status(err, res);
+      .then((res) => {
+        iExpect.htmlWith200Status(res);
 
         expect(res).to.redirect;
         expect(res.req.path).to.equal(`${constants.SITE_ROOT}/`);
-        done();
-      });
+      })
+      .catch((err) => { throw err; });
```
[the remark this answers is not in the document]
Comment (st3v3nhunt) [reply test/integration/app.js]: I think I'd prefer to use promises rather than async, await.

File: test/integration/bankHolidayAlert.js (hunk 1 above)
Comment (markysoft) [reply test/integration/bankHolidayAlert.js]: I see, so it's the lack of the return of the 'chai.request()' promise that is causing the silent failure.
In that case, all tests should return the chai.request(), and the catch-throw isn't necessary so can be removed from the tests.

Comment (st3v3nhunt) [reply test/integration/bankHolidayAlert.js]: Not quite as `chai.request` is being returned implicitly. I'll finish the changes to use async, it looks quite a bit better.

Comment (markysoft) [reply test/integration/bankHolidayAlert.js]: That's the theory, but the 'unknown page' test below fails silently with no return from chai.request, but correctly reports the test as failing if you **do** return the value.

File: test/integration/app.js [hunk 3]
```diff
@@ -1,42 +1,36 @@
...
-        const $ = cheerio.load(res.text);
-
-        expect($('.local-header--title--question').text().trim())
-          .to.equal('Page not found');
-        expect($('title').text()).to.equal('Page not found - NHS.UK');
-      })
-      .catch((err) => { throw err; });
+  it('should return a 404', async () => {
+    try {
+      await chai.request(server).get(`${constants.SITE_ROOT}/not-known`);
```
Comment (neilbmclaughlin) [inline test/integration/app.js]: Should there be an `expect.fail()` here for if no exception is thrown?

Comment (neilbmclaughlin) [reply test/integration/app.js]: Or, possibly, use expect.throw(..)

**Pull request merged**